src/de.rs

```diff
@@ -1451,13 +1454,27 @@ impl<'a> Deserializer<'a> {
                         expected: "a value",
                         found: token.1.describe(),
                     },
-                ))
+                ));
             }
             None => return Err(self.eof()),
         };
         Ok(value)
     }
 
+    fn parse_keylike(&mut self, at: usize, span: Span, key: &'a str) -> Result<Value<'a>, Error> {
+        let lowercase = key.to_ascii_lowercase();
+        let lowercase = lowercase.as_str();
+        if matches!(lowercase, "inf") || matches!(lowercase, "nan") {
```

Comment:
Can this not use `matches!` and use `==` instead?

Reply:
Was probably a leftover from some experiments, it should definitely use ==.